homeassistant/components/thermostat/heat_control.py

```diff
@@ -34,27 +37,34 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     name = config.get(CONF_NAME, DEFAULT_NAME)
     heater_entity_id = config.get(CONF_HEATER)
     sensor_entity_id = config.get(CONF_SENSOR)
+    min_temp = util.convert(config.get(CONF_MIN_TEMP), float, None)
+    max_temp = util.convert(config.get(CONF_MAX_TEMP), float, None)
+    target_temp = util.convert(config.get(CONF_TARGET_TEMP), float, None)
 
     if None in (heater_entity_id, sensor_entity_id):
         _LOGGER.error('Missing required key %s or %s', CONF_HEATER,
                       CONF_SENSOR)
         return False
 
-    add_devices([HeatControl(hass, name, heater_entity_id, sensor_entity_id)])
+    add_devices([HeatControl(hass, name, heater_entity_id, sensor_entity_id,
+                             min_temp, max_temp, target_temp)])
 
 
 # pylint: disable=too-many-instance-attributes
 class HeatControl(ThermostatDevice):
     """ Represents a HeatControl device. """
-
-    def __init__(self, hass, name, heater_entity_id, sensor_entity_id):
+    # pylint: disable=too-many-arguments
+    def __init__(self, hass, name, heater_entity_id, sensor_entity_id,
+                 min_temp, max_temp, target_temp):
```

Comment:
Just curious, why do you make min_temp, max_temp, and target_temperature required arguments since not all thermostats support that functionality? Why not set those to `None` in the function definition?

Clearly I'm missing something here.